optuna/integration/sklearn.py

```diff
@@ -0,0 +1,925 @@
+
+        self._check_is_fitted()
+
+        return self.study_.trials_dataframe
+
+    def __init__(
+        self,
+        estimator,  # type: BaseEstimator
+        param_distributions,  # type: Mapping[str, distributions.BaseDistribution]
+        cv=5,  # type: Optional[Union[BaseCrossValidator, int]]
+        enable_pruning=False,  # type: bool
+        error_score=np.nan,  # type: Union[float, str]
+        max_iter=1000,  # type: int
+        n_jobs=1,  # type: int
+        n_trials=10,  # type: int
+        random_state=None,  # type: Optional[Union[int, np.random.RandomState]]
+        refit=True,  # type: bool
+        return_train_score=False,  # type: bool
+        scoring=None,  # type: Optional[Union[Callable[..., float], str]]
+        study=None,  # type: Optional[study_module.Study]
+        subsample=1.0,  # type: Union[float, int]
+        timeout=None,  # type: Optional[float]
+        verbose=0  # type: int
+    ):
+        # type: (...) -> None
+
+        _check_sklearn_availability()
+
+        self.cv = cv
+        self.enable_pruning = enable_pruning
+        self.error_score = error_score
+        self.estimator = estimator
+        self.max_iter = max_iter
+        self.n_trials = n_trials
+        self.n_jobs = n_jobs
+        self.param_distributions = param_distributions
+        self.random_state = random_state
+        self.refit = refit
+        self.return_train_score = return_train_score
+        self.scoring = scoring
+        self.study = study
+        self.subsample = subsample
+        self.timeout = timeout
+        self.verbose = verbose
+
+    def _check_is_fitted(self):
+        # type: () -> None
+
+        attributes = [
+            'n_splits_',
+            'sample_indices_',
+            'scorer_',
+            'study_'
+        ]
+
+        if self.refit:
+            attributes += ['best_estimator_', 'refit_time_']
+
+        check_is_fitted(self, attributes)
+
+    def _check_params(self):
+        # type: () -> None
+
+        if not hasattr(self.estimator, 'fit'):
+            raise ValueError(
+                'estimator must be a scikit-learn estimator.'
+            )
+
+        if type(self.param_distributions) is not dict:
+            raise ValueError('param_distributions must be a dictionary.')
+
+        for name, distribution in self.param_distributions.items():
+            if not isinstance(distribution, distributions.BaseDistribution):
+                raise ValueError(
+                    'Value of {} must be a optuna distribution.'.format(name)
+                )
+
+        if self.enable_pruning and not hasattr(self.estimator, 'partial_fit'):
+            raise ValueError('estimator must support partial_fit.')
+
+        if self.max_iter <= 0:
+            raise ValueError(
+                'max_iter must be > 0, got {}.'.format(self.max_iter)
+            )
+
+        if self.study is not None and self.study.direction != 'maximize':
```

Comment:
Please replace `'maximize'` with `structs.StudyDirection.MAXIMIZE`. Otherwise, the following case fails.

```python
    study = optuna.create_study(direction='maximize')
    optuna_search = integration.OptunaSearchCV(
        ...
        study=study,
    )

    optuna_search.fit(X, y)
```

Reply:
done.